src/server/sched_func.c

```diff
@@ -335,5 +335,21 @@ set_sched_default(pbs_sched* psched) {
 		psched->sch_attr[(int) SCHED_ATR_sched_cycle_len].at_flags = ATR_VFLAG_DEFLT
 				| ATR_VFLAG_SET | ATR_VFLAG_MODCACHE;
 	}
+	if ((psched->sch_attr[(int) SCHED_ATR_sched_state].at_flags & ATR_VFLAG_SET)
+			==0) {
+		psched->sch_attr[(int) SCHED_ATR_sched_state].at_val.at_str =  malloc(SC_STATUS_LEN + 1);
+		if ( psched->sch_attr[(int) SCHED_ATR_sched_state].at_val.at_str == '\0' )
```

Comment:
There is several things wrong with this line.  First, you're comparing a string to a character.  Are you trying to check against NULL?  If not, since you used malloc(), the memory will be garbage, not zeroed.  Lastly, you have a space after the ( and before the ) which need to be removed.